app/controllers/application_controller.rb

```diff
@@ -5,6 +5,103 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+
+  get '/' do
+    erb :'index'
+  end
+
+
+  get '/signup' do
+    erb :'users/create_user'
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:password] == "" || params[:email] == ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.